cmd/krel/cmd/sign_blobs.go

```diff
@@ -0,0 +1,214 @@
+			if strings.HasSuffix(file, ".sha256") || strings.HasSuffix(file, ".sha512") ||
+				strings.HasSuffix(file, ":") || strings.HasSuffix(file, ".docker_tag") ||
+				(strings.ContainsAny(file, "SHA256SUMS") && strings.ContainsAny(file, "README") &&
+					strings.ContainsAny(file, "Makefile")) {
+				continue
+			}
+			temp := strings.TrimPrefix(file, object.GcsPrefix)
+			err = gcsClient.CopyToLocal(file, tempDir)
+			if err != nil {
+				return fmt.Errorf("copying file to sign: %w", err)
+			}
+
+			bundle = append(bundle, signingBundle{
+				destinationPathToCopy: filepath.Dir(temp),
+				fileToSign:            filepath.Base(temp),
+				fileLocalLocation:     fmt.Sprintf("%s/%s", tempDir, filepath.Base(temp)),
+			})
+		}
+	} else {
+		// Local files
+		for _, arg := range args {
+			bundle = append(bundle, signingBundle{
+				fileLocalLocation: arg,
+				fileToSign:        filepath.Base(arg),
+			})
+		}
+	}
+
+	t := throttler.New(int(signOpts.maxWorkers), len(args))
+	for _, fileBundle := range bundle {
+		go func(fileBundle signingBundle) {
+			logrus.Infof("Signing %s...", fileBundle.fileToSign)
+			signerOpts := sign.Default()
+			signerOpts.Verbose = signOpts.verbose
+			signerOpts.Timeout = signOpts.timeout
+			signerOpts.PrivateKeyPath = signBlobOpts.privateKeyPath
+			signerOpts.PublicKeyPath = signBlobOpts.publicKeyPath
+
+			signerOpts.OutputCertificatePath = fmt.Sprintf("%s/%s.cert", signBlobOpts.outputPath, fileBundle.fileToSign)
+			signerOpts.OutputSignaturePath = fmt.Sprintf("%s/%s.sig", signBlobOpts.outputPath, fileBundle.fileToSign)
+			if signBlobOpts.outputPath == "" {
+				signerOpts.OutputCertificatePath = fmt.Sprintf("%s.cert", fileBundle.fileLocalLocation)
+				signerOpts.OutputSignaturePath = fmt.Sprintf("%s.sig", fileBundle.fileLocalLocation)
+			}
+
+			signer := sign.New(signerOpts)
+			_, err := signer.SignFile(fileBundle.fileLocalLocation)
+			if err != nil {
+				t.Done(fmt.Errorf("signing the file %s: %w", fileBundle.fileLocalLocation, err))
+				return
+			}
+			t.Done(nil)
+		}(fileBundle)
+
+		if t.Throttle() > 0 {
+			break
+		}
+	}
+	if err := t.Err(); err != nil {
+		return fmt.Errorf("signing the blobs: %w", err)
+	}
+
+	if isGSBucket {
+		logrus.Info("Copying Certificates and Signatures back to the bucket...")
+		for _, fileBundle := range bundle {
+			certSignFiles := fmt.Sprintf("%s/%s*", signBlobOpts.outputPath, fileBundle.fileToSign)
+			if signBlobOpts.outputPath == "" {
+				certSignFiles = fmt.Sprintf("%s*", fileBundle.fileLocalLocation)
+			}
+
+			logrus.Infof("Copying %s...", certSignFiles)
+			_, err = gcli.GSUtilOutput("cp", "-n", certSignFiles, fmt.Sprintf("%s%s", object.GcsPrefix, fileBundle.destinationPathToCopy))
+			if err != nil {
+				return fmt.Errorf("copying certificates and signatures to the bucket: %w", err)
+			}
+		}
+	}
+
+	logrus.Info("Done")
+	return nil
+}
+
+func validateSignBlobsArgs(args []string) error {
+	if strings.HasPrefix(args[0], object.GcsPrefix) {
```

Comment:
This will panic if there are no args provided, we have to check the `len(args)` as well.